### PR TITLE
add xgcm_transform_wrapper to upstream stash

### PR DESCRIPTION
**Pull request by sditkovsky**
Added `FutureWarning` for wrapper function. There were a bunch of comments in there about possible improvements that I kept in. Added tests. Added cf-xarray, xgcm to ci environments.

### CODE REVIEW

**Comment by sditkovsky** [comment]
3.7 is building fine, but 3.8 is throwing up an error when updating the test environment. I think the issue is with `numba`, because before that it ran through that step fine. Know of any issues with numba and 3.8? Maybe try with conda instead of Pip?

**Comment by jbusecke** [comment]
Oh most definitely use conda. Numba has very complex (non-python) dependencies. This is a prime example of when to use conda.

File: ci/environment-py3.7.yml [hunk 1]
```diff
@@ -10,9 +10,12 @@ dependencies:
   - cmip6_preprocessing
   - xarrayutils
   - regionmask
+  - xgcm
+  - cf_xarray
+  - numba
   ##############
-  - pytest
+  - Pytest
```
Comment (jbusecke) [inline ci/environment-py3.7.yml:17]: I have never seen this capitalized. Any reason why you changed that?

File: tests/test_upstream_stash.py [hunk 2]
```diff
@@ -0,0 +1,73 @@
+import pytest
```
Comment (jbusecke) [inline tests/test_upstream_stash.py:1]: Thanks for adding these test.

**Review by jbusecke — APPROVED** [review]
Looks good to me.

**Pull request merged**